python/pyarrow/parquet/core.py

```diff
@@ -1020,7 +1019,7 @@ def __init__(self, where, schema, filesystem=None,
             sink = where
         self._metadata_collector = options.pop('metadata_collector', None)
         engine_version = 'V2'
-        self.writer = _parquet.ParquetWriter(
+        self.writer = ParquetWriter(
```

Comment:
`ParquetWriter` is not imported above?

Reply:
Thanks @pitrou ! We have two definitions of `ParquetWriter` and wrongly missed that.

This still doesn't really solve the problem I am facing because if I avoid importing from `pyarrow.parquet` or `pyarrow.parquet.core` and directly import `pyarrow._parquet` I get the same error:
```
>>> import pyarrow._parquet
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "pyarrow/_parquet.pyx", line 1, in init pyarrow._parquet
    # Licensed to the Apache Software Foundation (ASF) under one
ModuleNotFoundError: No module named 'pyarrow.includes.libparquet'
```

Reply:
Perhaps try to clean any leftovers first? Something like `git clean -Xfd .` (be careful with it)

Reply:
Thanks @pitrou debugging the `_parquet.cpp` file gave me an idea and I found some non purely header definitions that made the import to fail. That issue has been solved, I'll continue fixing some other minor things and will mark the PR ready for review once all issues are fixed. Thanks again!